### PR TITLE
API-8799: Respond with 422 from 526 submission endpoint if Veteran has a particpant_id but is missing a birls_id

### PR DESCRIPTION
**Pull request by SlakrHakr**
## Description of change
Catch and respond with error if the Veteran somehow is in a state of having a participant_id but not having a birls_id. This shouldn't ever really happen as I understand it, but its not a state we can resolve if it somehow does.

## Original issue(s)
https://vajira.max.gov/browse/API-8799

### CODE REVIEW

**Comment by SlakrHakr** [comment]
Will rebase and set this to merge into master once the following PR has been merged:
https://github.com/department-of-veterans-affairs/vets-api/pull/7499

**Review by jw81 — APPROVED** [review]

**Review by jw81 — DISMISSED** [review]

File: modules/claims_api/app/controllers/claims_api/v1/application_controller.rb [hunk 1]
```diff
@@ -24,6 +24,11 @@ def fetch_aud
 
       def validate_veteran_identifiers
         return if target_veteran.participant_id.present? && target_veteran.birls_id.present?
+
+        if target_veteran.participant_id.present? && target_veteran.birls_id.blank?
+          raise ::Common::Exceptions::UnprocessableEntity.new(detail: 'No birls_id while participant_id present')
```
Comment (hinzed1127) [inline modules/claims_api/app/controllers/claims_api/v1/application_controller.rb:29]: The wording of this is a little weird to me. I would have an easier time interpreting this error if it was simply "birls_id is missing for the veteran" or something like that

**Review by hinzed1127 — APPROVED** [review]
Question on wording, but otherwise good 👍

File: modules/claims_api/app/controllers/claims_api/v1/application_controller.rb (hunk 1 above)
Comment (SlakrHakr) [reply modules/claims_api/app/controllers/claims_api/v1/application_controller.rb:29]: fair point regarding the wording, i just stole what va.gov is responding with:
https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/v0/mpi_users_controller.rb#L21

im not sure that "birls_id is missing for the veteran" gives the full picture though. a Veteran could be missing their birls_id and their participant_id and that would be fine. its specifically the situation where the Veteran has a participant_id but is missing a birls_id

**Review by jw81 — APPROVED** [review]

**Pull request merged**